ddp-workspace/projects/ddp-dsm-ui/src/app/phi-manifest/phi-manifest.component.ts

```diff
@@ -0,0 +1,112 @@
+        this.additionalMessage = null;
+        let jsonData: any[];
+        this.dsmService.getRealmsAllowed(Statics.MEDICALRECORD).subscribe({
+            next: data => {
+                jsonData = data;
+                jsonData.forEach((val) => {
+                    if (this.realm === val) {
+                        this.allowedToSeeInformation = true;
+                    }
+                });
+                if (!this.allowedToSeeInformation) {
+                    this.additionalMessage = 'You are not allowed to see information of the selected study at that category';
+                }
+            },
+            error: () => null
+        });
+    }
+
+    ngOnInit(): void {
+        if (sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null) {
+            this.realm = sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+            this.checkRight();
+        } else {
+            this.additionalMessage = 'Please select a PE-CGS study';
+        }
+        window.scrollTo(0, 0);
+    }
+
+    getPhiReport(): void {
+        this.additionalMessage = null;
+        this.errorMessage = null;
+        if (this.realm != null && this.participantId && this.sequencingOrderId) {
+            this.loading = true;
+            this.dsmService.getPhiReport(this.realm, this.participantId, this.sequencingOrderId).subscribe(
+                {// need to subscribe, otherwise it will not send!
+                next: data => {
+                    // console.info(`received: ${JSON.stringify(data, null, 2)}`);
+                    const phiManifestResponse = PhiManifestResponseModel.parse(data);
+                    this.loading = false;
+                    this.downloadManifest(phiManifestResponse);
+                },
+                error: err => {
+                    // console.log(err);
+                    if (err._body === Auth.AUTHENTICATION_ERROR) {
+                        this.auth.doLogout();
+                    }
+                    this.errorMessage = 'Error - Failed to download report</b>' + err.error;
+                    this.loading = false;
+                }
+            });
+        } else {
+            this.errorMessage = 'Please select a study and enter information';
+        }
+    }
+
+    public downloadManifest(phiManifestResponseModel: PhiManifestResponseModel): void {
+        if (phiManifestResponseModel.errorMessage) {
+            this.additionalMessage = 'Error - ' + phiManifestResponseModel.errorMessage;
+            return;
+        }
+        const phiManifestModel: PhiManifestModel = PhiManifestModel.parse(phiManifestResponseModel.data[1]);
+        const map: {}[] = [];
+        map.push(phiManifestModel.getReportAsMap(phiManifestResponseModel.data[0], phiManifestResponseModel.data[1]));
+        Utils.createCSV(
+            phiManifestResponseModel.data[0],
+            map,
+            'PHI_REPORT_PT_' + phiManifestResponseModel.ddpParticipantId + '_ORDER_' +
```

Comment:
The data should not check, since downloading this report is the final step in generating shared learnings, but I will add a date just in case it helps the user when later they want to find the files they downloaded